modules/internal/ChapelDistribution.chpl

```diff
@@ -581,7 +587,19 @@ module ChapelDistribution {
     // atomics are available
     var _arrAlias: BaseArr;    // reference to base array if an alias
     var pid:int = nullPid; // privatized ID, if privatization is supported
+
+    proc isSliceArrayView() param {
+      return false;
+    }
+
+    proc isRankChangeArrayView() param {
+      return false;
+    }
   
+    proc isReindexArrayView() param {
+      return false;
+    }
```

Comment:
I had (and I think still have?) a comment in the code defining these routines on array views asking that question but went the conceptually simpler (to me) route on day one to sprint to getting things working.  I can take a stab at removing these calls and relying on type information if you like.  I believe my hesitation was not knowing what I could expect from casting a concrete object by a generic class name (do you have experience with that in Chapel?)